frontend/src/queries/schema/schema-general.ts

```diff
@@ -2253,6 +2253,25 @@ export interface ExperimentExposureQuery extends DataNode<ExperimentExposureQuer
 }
 
 export interface ExperimentQueryResponse {
+    // Legacy fields
+    kind?: NodeKind.ExperimentQuery
+    insight?: Record<string, any>[]
+    metric?: ExperimentMetric
+    variants?: ExperimentVariantTrendsBaseStats[] | ExperimentVariantFunnelsBaseStats[]
+    probability?: Record<string, number>
+    significant?: boolean
+    significance_code?: ExperimentSignificanceCode
+    stats_version?: integer
+    p_value?: number
+    credible_intervals?: Record<string, [number, number]>
+
+    // New fields
+    baseline?: ExperimentStatsBase
+    variant_results?: ExperimentVariantResultFrequentist[] | ExperimentVariantResultBayesian[]
+}
```

Comment:
We can construct this type using LegacyExperimentQueryResponse and NewExperimentQueryResponse:

```ts
export type ExperimentQueryResponse = LegacyExperimentQueryResponse | Partial<NewExperimentQueryResponse>
```

We don't have to declare `NewExperimentQueryResponse`; we can just union on an anonymous type. But that will probably mess with pedantry.

It'll be nice if we can add some type guards too:

```ts
const isLegacyExperimentQueryResponse = (resp: ExperimentQueryResponse): response is LegacyExperimentQueryResponse => 'kind' in resp && kind === NodeKind.ExperimentQuery;
```

and maybe a comment on `ExperimentQueryResponse` explaining why the type exists

Reply:
Yeah, the `export type` simplification does not work with the conversion to pydantic classes 🤷 

We have type guards for what we need for now I think (`isLegacyExperimentResponse` and `isNewExperimentResponse`). I think maybe we can get away with that. Technically, the we should only have to work with `metricResults: CachedNewExperimentQueryResponse` and `legacyMetricResults: CachedLegacyExperimentQueryResponse` in our UI now 🤞